include/nfl/poly.hpp

```diff
@@ -151,6 +151,15 @@ class poly {
   void ntt_pow_phi() { base.ntt_pow_phi(*this);}
   void invntt_pow_invphi() { base.invntt_pow_invphi(*this); }
 
+  /* manual serializers
+  */
+  inline void serialize(std::ostream& outputstream) {
```

Comment:
inline is redundnat for inner classes. Serialize should be const. use _data.data() before the reinterpretcast looks safer to me.